hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java

```diff
@@ -229,6 +237,40 @@ public void testMultipleChunksInSingleWriteOp(int numChunks)
     validateContent(inputData, bucket, bucket.getKey(keyName));
   }
 
+  @Test
+  public void testECContainerKeysCount()
+      throws IOException, InterruptedException, TimeoutException,
+      AuthenticationException {
+    byte[] inputData = getInputBytes(1);
+    final OzoneBucket bucket = getOzoneBucket();
+    ContainerOperationClient containerOperationClient =
+        new ContainerOperationClient(conf);
+    List<ContainerInfo> containerInfos =
+        containerOperationClient.listContainer(1, 100);
+    Map<ContainerID, Long> containerKeys = new HashMap<>();
+    for (ContainerInfo info : containerInfos) {
+      containerKeys.put(info.containerID(),
+          containerKeys.getOrDefault(info.containerID(), 0L) + 1);
+    }
+
+    String keyName = UUID.randomUUID().toString();
+    try (OzoneOutputStream out = bucket.createKey(keyName, 4096,
+        new ECReplicationConfig(3, 2, ECReplicationConfig.EcCodec.RS,
+            chunkSize), new HashMap<>())) {
+      out.write(inputData);
+    }
+    OzoneKeyDetails key = bucket.getKey(keyName);
+    cluster.restartStorageContainerManager(true);
```

Comment:
I guess this restart is here so SCM gets a full container report to get the updated key counts in the containers? Rather than restarting SCM, what about setting the FCR interval to a very low value, eg, in the init() method

```
    conf.setTimeDuration(HDDS_HEARTBEAT_INTERVAL, 500, TimeUnit.MILLISECONDS);
    conf.setTimeDuration(HDDS_CONTAINER_REPORT_INTERVAL, 1, TimeUnit.SECONDS);
```

Then change the test to something like:

```
...
    OzoneKeyDetails key = bucket.getKey(keyName);
    long currentKeyContainerID =
        key.getOzoneKeyLocations().get(0).getContainerID();
    Long priorKeys = containerKeys.get(new ContainerID(currentKeyContainerID));
    long expectedKeys = (priorKeys != null ? priorKeys : 0) + 1L;

    GenericTestUtils.waitFor(() -> {
      try {
        return containerOperationClient
            .listContainer(currentKeyContainerID, 100)
            .get(0).getNumberOfKeys() == expectedKeys;
      } catch (IOException exception) {
        fail("Unexpected exception " + exception);
        return false;
      }
    }, 100, 10000);
    validateContent(inputData, bucket, bucket.getKey(keyName));
```

On my laptop the test completes in 2 - 3 seconds now, rather than 7. Its a small win, probably worth it. I think its a better test, as it relies on the contaier report to be received by an already running SCM.

Reply:
This make sense. I have updated the test with your suggestion.